docs/main/04-guides/04-writing-a-promise.mdx

```diff
@@ -0,0 +1,259 @@
+```
+
+Next, we will use the Kratix CLI to add a Promise Configure workflow to our database Promise, so these dependencies are
+installed when we install the Promise.
+
+```bash
+kratix update dependencies ./deps/ --image kratix-guide/database-promise-pipeline:v0.1.0
+```
+
+We also need to build and load this image into our local environment:
+
+```bash
+docker build -t kratix-guide/database-promise-pipeline:v0.1.0 workflows/promise/configure/dependencies/configure-deps
+kind load docker-image kratix-guide/database-promise-pipeline:v0.1.0 --name platform
+```
+
+:::info
+Always remember to load new images into your environment when using kind!
+:::
+
+### Resource Configure Workflows
+
+With our Promise dependencies in place, we are now ready to fulfill resource requests.
+When a user makes a request of this database Promise, the Resource Configure workflow will execute,
+and we will need to provision a Postgres database in this workflow.
+
+To start, we can use the Kratix CLI again to add a container to the Resource Configure workflow:
+```
+kratix add container resource/configure/database-configure --image kratix-guide/database-resource-pipeline:v0.1.0 --language python
+```
+
+Take a look at the file `workflows/resource/configure/database-configure/kratix-guide-database-resource-pipeline/scripts/pipeline.py`
+
+This is our Resource Configure script, and it's written in Python using the [Kratix Python SDK](/main/kratix-sdks/python).
+It has some print lines and let's replace the content, so it generates a Postgres database for us:
+
+```python
+import kratix_sdk as ks
+import yaml
+
+def main():
+    sdk = ks.KratixSDK()
+    resource = sdk.read_resource_input()
+    name = resource.get_name()
+    size = resource.get_value("spec.size")
+
+    manifest = {
+        "apiVersion": "acid.zalan.do/v1",
+        "kind": "postgresql",
+        "metadata": {"name": name, "namespace": "default"},
+        "spec": {
+        	"teamId": "kratix",
+			"enableLogicalBackup": True,
+        	"volume": {
+            	"size": size
+        	},
+        	"numberOfInstances": 2,
+                "users": {
+                     "team-a": ["superuser", "createdb"]
+                },
+        	"postgresql": {
+            	"version": "16"
+        	}
+		}
+    }
+    data = yaml.safe_dump(manifest).encode("utf-8")
+    sdk.write_output("database.yaml", data)
+
+    status = ks.Status()
+    status.set("teamId", "kratix")
+    sdk.write_status(status)
+
+
+if __name__ == '__main__':
+    main()
+```
+
+Let's build and load this image into our local environment:
+
+```bash
+docker build -t kratix-guide/database-resource-pipeline:v0.1.0 workflows/resource/configure/database-configure/kratix-guide-database-resource-pipeline
+kind load docker-image kratix-guide/database-resource-pipeline:v0.1.0 --name platform
+```
+
+:::info
+Always remember to load new images into your environment when using kind!
```

Comment:
I definitely agree with the repetition, but maybe because we're repeating it, it could be nice to add an "Again, always remember to load new images...". This is optional, though, feel free to disregard it if you don't think it adds value.